internal/clock/simulated_clock_test.go

```diff
@@ -0,0 +1,279 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package clock
+
+import (
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+var (
+	// A non-zero reference time for tests
+	referenceTime    = time.Date(2020, time.January, 1, 12, 0, 0, 0, time.UTC)
+	shortTestTimeout = 10 * time.Millisecond // For non-blocking channel checks
+	fireTestTimeout  = 50 * time.Millisecond // When expecting a channel to fire
+)
+
+// Helper to assert that a channel receives a specific time within a timeout.
+func assertReceivesTime(t *testing.T, ch <-chan time.Time, expectedTime time.Time, timeout time.Duration, msgAndArgs ...interface{}) {
+	t.Helper()
+	select {
+	case actualTime := <-ch:
+		assert.True(t, expectedTime.Equal(actualTime), "Received time %v, expected %v. %s", actualTime, expectedTime, msgAndArgs)
+	case <-time.After(timeout): // Using real time.After for test timeout
+		t.Fatalf("Timeout waiting for time on channel. Expected %v. %s", expectedTime, msgAndArgs)
+	}
+}
+
+// Helper to assert that a channel does NOT receive a time within a short duration.
+func assertNotReceivesTime(t *testing.T, ch <-chan time.Time, timeout time.Duration, msgAndArgs ...interface{}) {
```

Comment:
Should we move this select block to the main test? I don't think we need a separate method for this.